primitives/io/src/lib.rs

```diff
@@ -128,26 +99,81 @@ pub trait Storage {
 		})
 	}
 
-	/// Get `key` from child storage, placing the value into `value_out` and return the number
-	/// of bytes that the entry in storage has beyond the offset or `None` if the storage entry
-	/// doesn't exist at all.
-	/// If `value_out` length is smaller than the returned length, only `value_out` length bytes
-	/// are copied into `value_out`.
+	/// Set `key` to `value` in the storage.
+	fn set(&mut self, key: &[u8], value: &[u8]) {
+		self.set_storage(key.to_vec(), value.to_vec());
+	}
+
+	/// Clear the storage of the given `key` and its value.
+	fn clear(&mut self, key: &[u8]) {
+		self.clear_storage(key)
+	}
+
+	/// Check whether the given `key` exists in storage.
+	fn exists(&self, key: &[u8]) -> bool {
+		self.exists_storage(key)
+	}
+
+	/// Clear the storage of each key-value pair where the key starts with the given `prefix`.
+	fn clear_prefix(&mut self, prefix: &[u8]) {
+		Externalities::clear_prefix(*self, prefix)
+	}
+
+	/// "Commit" all existing operations and compute the resulting storage root.
+	///
+	/// The hashing algorithm is defined by the `Block`.
+	///
+	/// Returns the SCALE encoded hash.
+	fn root(&mut self) -> Vec<u8> {
+		self.storage_root()
+	}
+
+	/// "Commit" all existing operations and get the resulting storage change root.
+	/// `parent_hash` is a SCALE encoded hash.
+	///
+	/// The hashing algorithm is defined by the `Block`.
 	///
-	/// See `child_get` for common child api parameters.
+	/// Returns an `Some(_)` which holds the SCALE encoded hash or `None` when
+	/// changes trie is disabled.
+	fn changes_root(&mut self, parent_hash: &[u8]) -> Option<Vec<u8>> {
+		self.storage_changes_root(parent_hash)
+			.expect("Invalid `parent_hash` given to `changes_root`.")
+	}
+
+	/// Get the next key in storage after the given one in lexicographic order.
+	fn next_key(&mut self, key: &[u8]) -> Option<Vec<u8>> {
+		self.next_storage_key(&key)
+	}
+
+
+	/// Deprecated, please use dedicated runtime apis.
+	fn child_get(
```

Comment:
Is this code used somewhere? I don't think so? 

So please remove it, the same goes for all deprecated unused functions.

Reply:
I did remove them first, but restore them back to retain compatibility with edgeware chain, but I would also really prefer to remove it (and the associated code).
Think we can break that?

Reply:
Yeah, if we not do it now, we need to keep this code forever.

Reply:
I did remove the code (it messed up my last master merge and it is in latest master merge commit).